ocs_ci/framework/pytest_customization/marks.py

```diff
@@ -392,6 +393,46 @@
     ),
     reason="Test runs ONLY on Fusion HCI provider and client clusters",
 )
+
+
+# when run_on_all_clients marker is used, there needs to be added cluster_index
+# parameter to the test to prevent any issues with the test parametrization
+def setup_multicluster_marker(marker_base, push_missing_configs=False):
+    """
+    Set up multicluster marker with parametrization based on client indexes.
+
+    Args:
+        marker_base: Base pytest marker to be parametrized
+        push_missing_configs: Boolean flag to push missing configs
+
+    Returns:
+        Parametrized marker or original marker if setup fails
+    """
+    try:
+        if push_missing_configs:
+            from ocs_ci.deployment.hosted_cluster import hypershift_cluster_factory
+
+            hypershift_cluster_factory(
```

Comment:
probably unrelated to this PR, but the concept of ODF client clusters is not tightly coupled to HCP hosted cluster. ODF client can be installed on non HCP hosted OCP cluster

Reply:
This concept was started few years back, I did not think about agent and external cluster having a role as client. Purpose then, when it was new was to make object entities of Cluster that aggregates capabilities of Hypershift OCP cluster, deployment helper functions and ODF client

Reply:
we have moved from this draft to https://github.com/red-hat-storage/ocs-ci/pull/12041, it is updated and verified